packages/cadl-python/src/emitter.ts

```diff
@@ -107,7 +106,9 @@ export async function $onEmit(program: Program, options: EmitterOptions) {
     if (program.compilerOptions.diagnosticLevel === "debug") {
         commandArgs.push("--debug");
     }
-    execFileSync(process.execPath, commandArgs);
+    if (!program.compilerOptions.noEmit && !program.hasError()) {
```

Comment:
Would just add the todo to make sure this either gets removed or implemented correctly whenever https://github.com/microsoft/cadl/issues/401 gets a solution.